controllers/snapshot/snapshot_adapter.go

```diff
@@ -154,11 +155,20 @@ func (a *Adapter) EnsureGlobalCandidateImageUpdated() (reconciler.OperationResul
 		for _, component := range a.snapshot.Spec.Components {
 			if component.Name == a.component.Name {
 				a.component.Spec.ContainerImage = component.ContainerImage
-				err := a.client.Patch(a.context, a.component, patch)
-				if err != nil {
-					a.logger.Error(err, "Failed to update Global Candidate for the Component",
+				if reflect.ValueOf(component.Source).IsValid() {
+					a.component.Status.LastBuiltCommit = component.Source.GitSource.Revision
+					err := a.client.Status().Patch(a.context, a.component, patch)
+					if err != nil {
+						a.logger.Error(err, "Failed to update the Status section of the Component while updating Global Candidate for the Component",
+							"Component.Name", a.component.Name)
+						return reconciler.RequeueWithError(err)
+					}
+				}
+				err1 := a.client.Patch(a.context, a.component, patch)
```

Comment:
I think it would be a good idea to move this patch call before we start checking if the status should be updated (so around line 157) since updating the `component.containerImage` is the more crucial operation of the two.

Reply:
Thank you for your help!